process/tags_v2.go

```diff
@@ -169,7 +170,11 @@ func unsafeIterateV2(buffer []byte, tagIndex int, cb func(i, total int, tag []by
 
 	footerBuffer := buffer[idx:]
 	footerIndex := 0
-
+	
+	if len(footerBuffer[footerIndex:]) < LenUint16 {
+		return
+	}
+	
```

Comment:
We also need bounds checking within the loop to handle the problem where the footer buffer is malformed

Reply:
done :)